From dacde1f11aa4fcbf7571ea520e7b1b8ccee154ec Mon Sep 17 00:00:00 2001
From: Dimitri John Ledkov <dimitri.ledkov@surgut.co.uk>
Date: Wed, 25 Jun 2025 17:42:39 +0100
Subject: [PATCH] cmake: add curve_keygen binary

When sodium is enabled, also build curve_keygen binary. This is to
bring cmake builds to parity with autoconf.

Fixes: https://github.com/zeromq/libzmq/issues/4675
---
 CMakeLists.txt | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 3ab2259e..7caf2c87 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1915,3 +1915,9 @@ if(ENABLE_NO_EXPORT)
   message(STATUS "Building with empty ZMQ_EXPORT macro")
   add_definitions(-DZMQ_NO_EXPORT)
 endif()
+
+if (ENABLE_CURVE)
+  add_executable(curve_keygen tools/curve_keygen.cpp)
+  target_link_libraries(curve_keygen libzmq)
+  install(TARGETS curve_keygen RUNTIME DESTINATION bin)
+endif()
-- 
2.48.1

